frappe/templates/includes/comments/comments.py

```diff
@@ -18,19 +18,23 @@
 @frappe.whitelist(allow_guest=True)
 @rate_limit(key="reference_name", limit=get_comment_limit, seconds=60 * 60)
 def add_comment(comment, comment_email, comment_by, reference_doctype, reference_name, route):
-	doc = frappe.get_doc(reference_doctype, reference_name)
+	if frappe.session.user == "Guest":
+		if reference_doctype not in ("Blog Post", "Web Page"):
+			return
+
+		if not frappe.db.get_single_value("Blog Settings", "allow_guest_to_comment"):
+			return
 
-	if frappe.session.user == "Guest" and doc.doctype not in ["Blog Post", "Web Page"]:
-		return
+		if frappe.db.exists("User", comment_email):
+			frappe.throw(_("Please login to post a comment."))
 
 	if not comment.strip():
-		frappe.msgprint(_("The comment cannot be empty"))
-		return False
+		frappe.throw(_("The comment cannot be empty"))
 
 	if URLS_COMMENT_PATTERN.search(comment) or EMAIL_PATTERN.search(comment):
-		frappe.msgprint(_("Comments cannot have links or email addresses"))
-		return False
```

Comment:
This might be by design?

Reply:
Doesn't look like this value is used 🤔 

https://github.com/frappe/frappe/blob/1cbb60adf93a741827f50ac0c54bb9f1bd1494ba/frappe/templates/includes/comments/comments.html#L172-L192

Reply:
Let me check once.

Reply:
You're right, it's on purpose